plugins/searchdomains/plugin.go

```diff
@@ -0,0 +1,68 @@
+// Copyright 2018-present the CoreDHCP Authors. All rights reserved
+// This source code is licensed under the MIT license found in the
+// LICENSE file in the root directory of this source tree.
+
+package searchdomains
+
+// This is an searchdomains plugin that adds default DNS search domains.
+
+import (
+	"github.com/coredhcp/coredhcp/handler"
+	"github.com/coredhcp/coredhcp/logger"
+	"github.com/coredhcp/coredhcp/plugins"
+	"github.com/insomniacslk/dhcp/dhcpv4"
+	"github.com/insomniacslk/dhcp/dhcpv6"
+	"github.com/insomniacslk/dhcp/rfc1035label"
+)
+
+var log = logger.GetLogger("plugins/searchdomains")
+
+// Plugin wraps the default DNS search domain options.
+// Note that importing the plugin is not enough to use it: you have to
+// explicitly specify the intention to use it in the `config.yml` file, in the
+// plugins section. For searchdomains:
+//
+// server6:
+//   listen: '[::]547'
+//   - searchdomains: domain.a domain.b
+//   - server_id: LL aa:bb:cc:dd:ee:ff
+//   - file: "leases.txt"
+//
+var Plugin = plugins.Plugin{
+	Name:   "searchdomains",
+	Setup6: setup6,
+	Setup4: setup4,
+}
+
+// These are the DHCP options that are added by the plugin.
+// Note that DHCPv4 and DHCPv6 options are totally independent.
+// If you need the same settings for both, you'll need to configure
+// this plugin once for the v4 and once for the v6 server.
+var v4Option dhcpv4.Option
+var v6Option dhcpv6.Option
```

Comment:
Using global variables for the returned options like this means that if any other plugin modifies the packet in-place, it would cause side-effects for every other packet handled by this plugin.
You'll need to deep-copy the option on every packet in the handler instead, or keep the search list instead of the full option and build the option by `copy`ing the list in the handlers

(for v4 the option is actually serialized to bytes during `UpdateOption` so it'll work, but it's not an API guarantee afaik. For v6 it's not copied at all)

Reply:
Wow, I didn't consider that. I switched to storing the search list in memory and copying it into a new `Option` each time, per your suggestion.